### ISSUE TITLE
gcc static analyzer analyzes sqlite infinitely long

### ISSUE THREAD

**Issue by dkrupp** [Ericsson/codechecker#4059]
`CodeChecker analyze ./compile_commands.json --analyzers gcc -e gcc -c -o reports_gcc`

The analysis of the following files were problematic
```
sqlite/shell.c
sqlite/sqlite3.c
```
memory consumption is stellar during anlaysis.
Gcc version: cc-13 --version
gcc-13 (Ubuntu 13.1.0-8ubuntu1~20.04.2) 13.1.0

CodeChecker version: commit 5ebdd4d7f26267d3cd9d8900f19c37aef2f15843 

Please check if we can get rid off this by disabling some checkers.

**Comment by whisperity** [comment]
`sqlite3.c` is a unity build source file composing the entire SQLite library as a single TU...

As per the [documentation](http://sqlite.org/src/dir?ci=trunk):

> The amalgamation source file is more than 200K lines long. Some symbolic debuggers (most notably MSVC) are unable to deal with files longer than 64K lines. To work around this, a separate Tcl script, tool/split-sqlite3c.tcl, can be run on the amalgamation to break it up into a single small C file called sqlite3-all.c that does #include on about seven other files named sqlite3-1.c, sqlite3-2.c, ..., sqlite3-7.c. In this way, all of the source code is contained within a single translation unit so that the compiler can do extra cross-procedure optimization, but no individual source file exceeds 32K lines in length.

Although there is a `./configure --disable-amalgamation` flag, unfortunately, even using this flag didn't help in letting go of the `sqlite3.c` file during compilation...

**Comment by Szelethus** [comment]
I didn't find this to be the case.

When I ran GCC with only `gcc-double-free` enabled, the analysis of this file too 559s. When I enabled all GCC checkers, it took 1027s. This is rather long, but in comparison to how the proudly heavyweight clangsa performs on huge files, this is I think okay.
